src/protocol/tx_format/input.md

```diff
@@ -105,5 +108,6 @@ Transaction is invalid if:
 - `dataLength != len(data)`
 - `predicateLength * 4 != len(predicate)`
 - `predicateDataLength != len(predicateData)`
+- `predicateGasUsed > MAX_GAS_PER_PREDICATE`
```

Comment:
We should also have a check that total sum of `predicate.gasUsed` <= `tx.gasLimit`.

Reply:
this is now addressed in the transaction validity rules